apps/src/blockly/addons/functionEditor.js

```diff
@@ -237,7 +238,20 @@ export default class FunctionEditor {
     this.showForFunction(hiddenProcedure, procedureType);
   };
 
-  handleDelete() {
+  onDeletePressed() {
+    Blockly.customSimpleDialog({
+      bodyText: msg.confirmDeleteFunctionWarning({
+        functionName: this.block.getProcedureModel().getName(),
+      }),
+      cancelText: msg.delete(),
+      isDangerCancel: true,
+      confirmText: msg.keep(),
+      onConfirm: null,
+      onCancel: this.onDeleteConfirmed.bind(this),
```

Comment:
Is this backward? I'd expect `confirmText` and `onConfirm` to handle the deletion and `cancelText` and `onCancel` to handle bailing out of the deletion.

Also, why is there no handler for `onConfirm`?

Reply:
It looks like you had to swap them to retain the order of the buttons relative to CDO. I think it's worth, at least, leaving a comment about why it's the opposite of what one would expect. I also wonder if it's worth asking Mark whether the OG button ordering is best (this could be done in a follow-up PR).

Reply:
The reason for this is it allows us to have a red delete button and a purple keep button. I'll add a comment explaining this. The reason there is no `onConfirm` handler is we do nothing if they want to keep the function.

Reply:
Added comments. We can follow-up on if we should switch up the order/color, but I don't think that's blocking given no one has complained about the existing modal.

Reply:
I think the thing I was missing re: the handler was that the modal closes regardless of which button you push -- this is hidden in the modal code, presumably.